src/enzo/hydro_rk/EOS.h

```diff
@@ -143,6 +144,38 @@ inline void EOS(float &p, float &rho, float &e, float &h, float &cs, float &dpdr
 
   }
 
+  /* Only works in mode 1 and only updates internal energy */
+  if (eostype == 7) {
+    float lenu, denu, tu, velu, tempu;
+    GetUnits(&denu, &lenu, &tempu, &tu, &velu, 1);
+    double rho_cr = 1e-15;
+    double new_gamma = 0.0;
+    double c_s = EOSSoundSpeed;
+    rho_cr /= denu;
+    
+    if(rho/rho_cr <= 0.25) {
+      new_gamma = Gamma;
+      cs = c_s*pow(rho/rho_cr, new_gamma);
+    }
+    else if(rho/rho_cr <= 5.0) {
+      new_gamma = 1.1;
+      cs = c_s*pow(rho/rho_cr, new_gamma);
+    }
+    else {
+      new_gamma = 4.0/3.0;
+      cs = c_s*pow(rho/rho_cr, new_gamma);
+    }
+    
+    p = rho*cs*cs;
```

Comment:
Following equation 30 in Federrath et al. 2010, should this be p = rho* c_s * cs instead?

Reply:
Good catch. I confirmed that this should be true. This bug has been in the EOS since the start. I fixed it.